_posts/2018-12-05-multilanguage-build-system.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: posts
+title: "Bazel - a multilanguage build system"
+authors:
+  - laurentlb
+---
+
+Part of [Bazel vision](https://docs.bazel.build/versions/master/bazel-overview.html)
+is to create a multilanguage build system: it should be possible to seamlessly
+use and mix multiple languages in a codebase and to orchestrate the builds and
+the tests with a single build tool. As a codebase gets bigger, this feature
+becomes more and more important.
+
+When a user wants a build system to be multilanguage, they might mean:
+
+*   Is the build system extensible? In other words, can I add support for a new
+    language, even if it doesn't have much adoption yet, or if it is internal?
+*   How many languages are already supported?
+*   How big is the ecosystem?
+*   Does it provide a high level of quality for the languages I use?
+*   Can I use it in production?
+
+Bazel allows you to [create your own rules](https://docs.bazel.build/versions/master/skylark/concepts.html).
+More importantly, however, it allows rules to communicate and pass information
+using "providers." For example, if a rule depends on a C++ library, it might
+need to access the appropriate headers or the compiled library. This is
+extensible, because rules declare what they expect from their dependencies. Any
+rule implementing that interface (that is, returning the correct providers) will
+be accepted as a dependency. For this reason, the ecosystem can grow and you can
+create new rules that interact with existing rules.
+
+Many rules exist for Bazel. A company like Google has more than a thousand
+internal rules, created by many teams, sometimes just for a single project, to
+handle their specific needs. Many other rules are public and shared in places
+like GitHub. The [Awesome-Bazel](https://github.com/jin/awesome-bazel#rules)
+page lists Bazel rules for many languages and tools. If you use a well-known
+language, it's likely you'll find an existing rule for it.
+
+However, quantity doesn't imply quality. As with any open-source code published
+on the Internet, there is no guarantee that a repository is properly designed
+and maintained. To help users, we want to promote a set of high-quality rules
+that they can trust. Our website lists [a few external rules](
+https://docs.bazel.build/versions/master/be/overview.html#additional-rules), but
+we are working on a more principled process for recommendations - for example,
+we should recommend only rules that we have properly tested.
```

Comment:
Didn't you have a doc about how a good rule set looks like (I can't find it just now). If you have it, please link it here.

Reply:
This? https://docs.bazel.build/versions/master/bazel-vision.html#so-what-is-a-good-ruleset

There's already a link to the vision.

Reply:
ah, forgot that this is part of the vision doc. If you feel like it, then link to the specific subsection from here, but I can also live without